### PR TITLE
puller(ticdc): remove some lock from region_state.go

### LINKED ISSUE #7423: remove some lock from region_state.go

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #7423

### What is changed and how it works?
remove unnecessary locks from the region state
region is processed by a single goroutine or hashed to one goroutine,

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/kv/client.go [hunk 1]
```diff
@@ -1314,13 +1314,12 @@ func (s *eventFeedSession) sendResolvedTs(
...
 	resolvedTs *cdcpb.ResolvedTs,
 	worker *regionWorker,
-	addr string,
 ) error {
 	statefulEvents := make([]*regionStatefulEvent, worker.concurrency)
 	// split resolved ts
 	for i := 0; i < worker.concurrency; i++ {
 		// Allocate a buffer with 1.5x length than average to reduce reallocate.
-		buffLen := len(resolvedTs.Regions) / worker.concurrency * 3 / 2
+		buffLen := len(resolvedTs.Regions) / worker.concurrency * 2
```
Comment (amyangfei) [inline cdc/kv/client.go:1322]: The comment is not updated.

File: cdc/kv/client.go [hunk 2]
```diff
@@ -1314,13 +1314,12 @@ func (s *eventFeedSession) sendResolvedTs(
 	ctx context.Context,
 	resolvedTs *cdcpb.ResolvedTs,
 	worker *regionWorker,
-	addr string,
 ) error {
 	statefulEvents := make([]*regionStatefulEvent, worker.concurrency)
 	// split resolved ts
 	for i := 0; i < worker.concurrency; i++ {
 		// Allocate a buffer with 1.5x length than average to reduce reallocate.
```
Comment (asddongmen) [inline cdc/kv/client.go:1321]: Please revise this comment

File: cdc/kv/region_worker.go [hunk 3]
```diff
@@ -193,9 +193,9 @@ func (w *regionWorker) handleSingleRegionError(err error, state *regionFeedState
 		zap.String("namespace", w.session.client.changefeed.Namespace),
 		zap.String("changefeed", w.session.client.changefeed.ID),
 		zap.Uint64("regionID", regionID),
-		zap.Uint64("requestID", state.getRequestID()),
-		zap.Stringer("span", state.getRegionSpan()),
-		zap.Uint64("resolvedTs", state.getRegionInfoResolvedTs()),
+		zap.Uint64("requestID", state.requestID),
+		zap.Stringer("span", state.sri.span),
+		zap.Uint64("resolvedTs", state.sri.resolvedTs),
```
Comment (amyangfei) [inline cdc/kv/region_worker.go:198]: How can we guarantee there is not data race with `state.sri.resolvedTs` after removing lock.

Comment (sdojjy) [reply cdc/kv/region_worker.go:198]: the value of the `state.sri` never be changed after the state is created?

File: cdc/kv/client.go (hunk 2 above)
Comment (sdojjy) [reply cdc/kv/client.go:1321]: done

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**